features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Create/Default/APICreateDefault.jsx

```diff
@@ -70,6 +70,9 @@ function APICreateDefault(props) {
             }
         });
     }, []);
+    const [isRevisioning, setIsRevisioning] = useState(false);
+    const [isDeploying, setIsDeploying] = useState(false);
+    const [isPublishButtonClicked, setIsPublishButtonClicked] = useState(false);
```

Comment:
Why we need another state here? 
I think we can use the `isCreating` state to identify whether the user has clicked on either Create or Create & Publish button